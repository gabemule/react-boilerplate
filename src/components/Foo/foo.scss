.foo {
  background-color: #d2d2d2;
}
