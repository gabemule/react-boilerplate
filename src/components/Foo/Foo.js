import React from "react";

import "./foo.scss";
import PropTypes from "prop-types";

const Foo = ({children}) => {
  return (
    <div className={"foo"}>
      {children}
    </div>
  );
};

Foo.propTypes = {
  children: PropTypes.any,
};

export default Foo;
