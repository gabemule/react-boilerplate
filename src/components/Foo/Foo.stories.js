import React from 'react'
import { Foo } from '@/components/Foo'

const componentProps = {
	title: 'Components/Foo',
	component: Foo
}

export default componentProps

const Template = (args) => <Foo {...args}> Foo Component </Foo>

export const Default = Template.bind({})
