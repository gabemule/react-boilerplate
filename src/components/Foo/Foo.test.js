import React from "react";
import { shallow } from "enzyme";

// Components
import Foo from "./Foo";

// Constants
const FOO_COMPONENT_TEXT = "Foo Component";
const FOO_COMPONENT_CLASS = ".foo";

/**
 * Setup method to avoid code repeat
 * @param props
 * @returns {{wrapper, props}}
 */
function setup(props) {
  const wrapper = shallow(<Foo>{FOO_COMPONENT_TEXT}</Foo>);
  return { wrapper, props };
}

describe("Foo Test Suite", () => {
  it("Should match snapshot", () => {
    expect(Foo).toMatchSnapshot();
  })
  it("Should have class", () => {
    const { wrapper } = setup();
    expect(wrapper.find(FOO_COMPONENT_CLASS).exists()).toBe(true);
  });
  it("Should have text", () => {
    const { wrapper } = setup();
    expect(wrapper.find(FOO_COMPONENT_CLASS).text()).toBe(FOO_COMPONENT_TEXT);
  });
});
