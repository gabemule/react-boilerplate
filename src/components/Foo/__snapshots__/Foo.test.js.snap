// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foo Test Suite Should match snapshot 1`] = `[Function]`;
